transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -777,19 +720,24 @@ public ChannelHandler removeLast() {
             assert ctx != null;
 
             if (!inEventLoop) {
-                try {
-                    executor.execute(() -> remove0(ctx));
-                    return ctx.handler();
-                } catch (Throwable cause) {
-                    handlers.add(idx, ctx);
-                    throw cause;
-                }
+                return scheduleRemove(idx, ctx);
             }
         }
         remove0(ctx);
         return ctx.handler();
     }
 
+    @SuppressWarnings("unchecked")
+    private <T extends ChannelHandler> T scheduleRemove(int idx, DefaultChannelHandlerContext ctx) {
+        try {
+            ctx.executor().execute(() -> remove0(ctx));
+            return (T) ctx.handler();
+        } catch (Throwable cause) {
+            handlers.add(idx, ctx);
+            throw cause;
```

Comment:
How does this compile?

Reply:
@carl-mastrangelo the compiler is smart enough when using Java8 to understand that the code in the `try` block can only throw an `unchecked exception` and so we will only rethrow this.